apps/trading/client-pages/markets/use-markets-stats.tsx

```diff
@@ -0,0 +1,96 @@
+import type { MarketMaybeWithCandles } from '@vegaprotocol/markets';
+import { AccountType } from '@vegaprotocol/types';
+import { addDecimal, priceChangePercentage } from '@vegaprotocol/utils';
+import { orderBy } from 'lodash';
+
+/**
+ * useTotalVolume24hCandles returns 24 hr candles with total volume
+ * this is used to draw the sparkline
+ *
+ * @param activeMarkets
+ * @returns
+ */
+export const useTotalVolume24hCandles = (
+  activeMarkets: MarketMaybeWithCandles[] | null
+) => {
+  const candles = [];
+  if (!activeMarkets || activeMarkets.length === 0) return [];
+  for (let i = 0; i < 24; i++) {
+    const totalVolume24hr = activeMarkets.reduce((acc, market) => {
+      const c = market.candles?.[i];
+      if (!c) return acc;
+      return (
+        acc +
+        Number(
+          addDecimal(
+            c.notional,
+            market.decimalPlaces + market.positionDecimalPlaces
+          )
+        )
+      );
+    }, 0);
+    candles.push(totalVolume24hr);
+  }
+  return candles;
+};
+
+/**
+ * useTopGainers returns the top 3 markets with highest gains, i.e. sorted by biggest 24h change
+ *
+ * @param activeMarkets
+ * @returns MarketMaybeWithCandles[]
+ */
+export const useTopGainers = (
+  activeMarkets: MarketMaybeWithCandles[] | null
+): MarketMaybeWithCandles[] => {
+  return orderBy(
+    activeMarkets,
+    [
+      (m) => {
+        if (!m.candles?.length) return 0;
+        return Number(
+          priceChangePercentage(
+            m.candles.filter((c) => c.close !== '').map((c) => c.close)
+          )
+        );
+      },
+    ],
+    ['desc']
+  ).slice(0, 3);
+};
+
+/**
+ * useNewListings returns the top 3 markets with the most recent opening timestamps on the network
+ *
+ * @param activeMarkets
+ * @returns MarketMaybeWithCandles[]
+ */
+export const useNewListings = (
+  activeMarkets: MarketMaybeWithCandles[] | null
+): MarketMaybeWithCandles[] => {
+  return orderBy(
+    activeMarkets,
+    [(m) => new Date(m.marketTimestamps.open).getTime()],
+    ['desc']
+  ).slice(0, 3);
+};
+
+/**
+ * useTVL returns the total value locked in the network
+ * by summing the balances of all insurance accounts
+ *
+ * @param activeMarkets
+ * @returns MarketMaybeWithCandles[] | null
+ */
+export const useTVL = (activeMarkets: MarketMaybeWithCandles[] | null) => {
```

Comment:
this is calculated by adding the Insurance Pool Balance for each market. 
how can I convert this to quantum?


note: this works well for perps/futures, not sure for spots.

Reply:
<img width="908" alt="image" src="https://github.com/vegaprotocol/frontend-monorepo/assets/16125548/4dfe10da-498a-420f-b714-262c989a7608">
we also have this, can we use it for this